Sources/Tracking/CustomerIO.swift

```diff
@@ -238,6 +246,59 @@ public class CustomerIO: CustomerIOInstance {
             )
     }
 
+    // Swizzled method for APN device token.
+    @objc
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let token = String(apnDeviceToken: deviceToken)
+        Self.shared.registerDeviceToken(token)
+    }
+
+    // Implement method swizzling for APN device token. Swizzling will ensure that the
+    // device token gets registered without needing the developer to add call respective
+    // Customer.io SDK methods to register the device.
+    @available(iOSApplicationExtension, unavailable)
+    func swizzleDidReceiveRemoteNotification() {
+        DispatchQueue.main.async {
+            let appDelegate = UIApplication.shared.delegate
+            let appDelegateClass: AnyClass? = object_getClass(appDelegate)
+            let originalSelector = #selector(UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+            let swizzledSelector = #selector(self.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+            guard let swizzledMethod = class_getInstanceMethod(CustomerIO.self, swizzledSelector) else {
+                return
+            }
+            if let originalMethod = class_getInstanceMethod(appDelegateClass, originalSelector) {
+                method_exchangeImplementations(originalMethod, swizzledMethod)
+            }
+        }
+    }
+
+    // Swizzled method for FCM Token. This method would not be called if the app is using APNS.
+    @objc
+    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+        if let token = fcmToken {
+            Self.shared.registerDeviceToken(token)
+        }
+    }
+
+    // Implement method swizzling for FCM token. Swizzling the FCM token method will ensure
+    // that the token is registered with CIO even when the developer doesn't call the respective
+    // CIO method after token generation.
+    @available(iOSApplicationExtension, unavailable)
+    func swizzleDidReceiveFCMRegistrationToken() {
+        DispatchQueue.main.async {
+            let messagingDelegate = Messaging.messaging().delegate
+            let messagingDelegateClass: AnyClass? = object_getClass(messagingDelegate)
+            let swizzledSelector = #selector(self.messaging(_:didReceiveRegistrationToken:))
+            let originalSelector = #selector(MessagingDelegate.messaging(_:didReceiveRegistrationToken:))
+            guard let swizzledMethod = class_getInstanceMethod(CustomerIO.self, swizzledSelector) else {
+                return
+            }
+            if let originalMethod = class_getInstanceMethod(messagingDelegateClass, originalSelector) {
+                method_exchangeImplementations(originalMethod, swizzledMethod)
+            }
+        }
+    }
```

Comment:
The `CustomerIO.swift` file is already a large file. I would prefer this code be moved to a different file. 

I suggest seeing how [the auto screenview tracking was done](https://github.com/customerio/customerio-ios/blob/main/Sources/Tracking/CustomerIOImplementation%2BScreenViews.swift) by using an `extension`, we are able to move the logic into another file.

Reply:
As mentioned in description, I would be cleaning this up and putting this in a separate file.